src/main/java/com/google/cloud/genomics/dataflow/functions/JoinNonVariantSegmentsWithVariants.java

```diff
@@ -194,44 +181,45 @@ public RetrieveFn(OfflineAuth auth, String fields) {
       }
 
       @Override
-      public void processElement(DoFn<StreamVariantsRequest, Iterable<Variant>>.ProcessContext context)
+      public void processElement(DoFn<StreamVariantsRequest, KV<KV<String, Long>, Iterable<Variant>>>.ProcessContext context)
           throws Exception {
+        StreamVariantsRequest request = context.element();
 
-        Iterator<StreamVariantsResponse> iter = VariantStreamIterator.enforceShardBoundary(auth, context.element(),
-            ShardBoundary.Requirement.NON_VARIANT_OVERLAPS, fields);
+        Iterator<StreamVariantsResponse> iter = VariantStreamIterator.enforceShardBoundary(auth, request,
+            ShardBoundary.Requirement.OVERLAPS, fields);
 
         if (iter.hasNext()) {
           // We do have some data overlapping this site.
           List<Iterable<Variant>> allVariantsForRequest = new ArrayList<>();
           while (iter.hasNext()) {
             allVariantsForRequest.add(iter.next().getVariantsList());
           }
-          context.output(Iterables.concat(allVariantsForRequest));
+          context.output(KV.of(KV.of(request.getReferenceName(), request.getStart()), Iterables.concat(allVariantsForRequest)));
         }
       }
     }
   }
 
-  public static final class CombineVariantsFn extends DoFn<Iterable<Variant>, Variant> {
+  public static final class CombineVariantsFn extends DoFn<KV<KV<String, Long>, Iterable<Variant>>, Variant> {
```

Comment:
Why the extra `KV<String, Long>`, and for what is would the extra KV in the object be useful as it expands each object?  What was wrong with having `Iterable<Variant>` from before?  It would be nice to have some JavaDoc documentation added to this function to provide the user some details regarding its purpose, and why it is built this way.

When adding the JavaDoc, it would be nice to say more than that the reference chromosome and start position is required, but more along the lines of why and how it would be used.